odbc/src/api/free_stmt_tests.rs

```diff
@@ -0,0 +1,137 @@
+mod unit {
+    use crate::{
+        handles::definitions::{
+            BoundColInfo, Connection, ConnectionState, Env, EnvState, MongoHandle, Statement,
+            StatementState,
+        },
+        map, SQLFreeStmt,
+    };
+    use bson::doc;
+    use definitions::{FreeStmtOption, Nullability, SqlReturn};
+    use mongo_odbc_core::json_schema::simplified::{Atomic, Schema};
+    use mongo_odbc_core::json_schema::BsonTypeName;
+    use mongo_odbc_core::{mock_query::MongoQuery, MongoColMetadata, MongoStatement};
+    use mongo_odbc_core::{Error, TypeMode};
+    use std::ptr::null_mut;
+
+    fn create_stmt_handle() -> *mut MongoHandle {
+        let env = &mut MongoHandle::Env(Env::with_state(EnvState::Allocated));
+        let conn =
+            &mut MongoHandle::Connection(Connection::with_state(env, ConnectionState::Allocated));
+        let stmt: *mut _ =
+            &mut MongoHandle::Statement(Statement::with_state(conn, StatementState::Allocated));
+
+        stmt
+    }
+
```

Comment:
related to the question of the default mongostatement.close_cursor, maybe we should add a test where we have a statement that isn't a MongoQuery and call close?

Reply:
done!